apps/checkout-app/frontend/components/elements/ClientProvider/ClientProvider.tsx

```diff
@@ -1,22 +1,31 @@
+import { API_URL } from "@/constants";
 import { useApp } from "@/frontend/hooks/useApp";
 import { getClient } from "@/frontend/misc/auth";
-import { useMemo } from "react";
+import { useEffect, useRef } from "react";
 import { Provider } from "urql";
 
 const ClientProvider: React.FC = ({ children, ...props }) => {
-  const app = useApp();
+  const { app } = useApp();
 
   const token = app?.getState()?.token;
 
-  const client = useMemo(() => {
-    if (token) {
-      return getClient(token);
+  const client = useRef(getClient(API_URL, token));
```

Comment:
The token should not be passed from a React view but rather the `app` should be imported and this:
```
      getAuth: getAuth(token),
```
should be:
```
      getAuth: () => app.getState().token,
```

Reply:
I've changed it to passing app to getClient and getting token as you suggested in getAuth